tuxemon/event/actions/set_monster_level.py

```diff
@@ -21,12 +21,12 @@
 
 from __future__ import annotations
 from tuxemon.event.eventaction import EventAction
-from typing import NamedTuple, final
+from typing import NamedTuple, final, Union
 
 
 class SetMonsterLevelActionParameters(NamedTuple):
-    slot: int
-    level: int
+    slot: Union[int, None]
```

Comment:
I think this is better typed as `Optional[int]`

Reply:
Usually yes, but here the type is being used to parse the parameter by trying to cast to each type sequentially in case of a Union (so the order matters). Maybe support for Optional could also be added, but for now, it is not.

Reply:
So this is fine as is?

Reply:
It should be, yes. We could try to accept also `Optional` in the parsing function, in a separate PR, and change every occurrence of `Union[whatever, None]` in event parameters. It is very low priority, though.

Reply:
ty for walking me through that vnmabus

Reply:
I've done some tests with Optional[int] based on this convo - in this action, the behaviour before and after was the same, both for valid params (int and none) and invalid params (float and str - they were accepted without raising an error in both the Union and Optional cases). 

I can make the change here if you want, as it seems the same, but I'll raise an issue seperately to track changing all the others.

Reply:
It seems that `Optional[whatever]` is translated at runtime to `Union[whatever, None]`. That is why it works. In our case, the order in which each type appears matters, as each type is tried sequentially, so we have to be 100% sure that future versions of Python do not change the order, and `None` is still the last parameter. If that is the case we could change them safely.

Reply:
I think you're right - I've looked at that code now (cast_values in tools.py) and if Optional was implemented in any other way, that code would fail. I don't think we want to rely on a quirk of the python implementation, so I'll keep it as Union and close the other issue I raised. Thanks for checking that!

Reply:
The official documentation says literally "Optional[X] is equivalent to Union[X, None]". I think that is unlikely that the behaviour changes in future versions, so if you want to use optional instead, it is ok for me.

Reply:
I'm happy either way - Union[X, None] reminds users that they need to handle both the X and None cases, which is good. But if Optional is used everywhere else in the code base then it might be better to be consistent. So I don't mind.

It's good to know that the choice is just a style choice, and both will have the same effect behind the scenes, I think that's the important thing.